srcts/src/bindings/input/date.ts

```diff
@@ -257,7 +258,7 @@ class DateInputBinding extends DateInputBindingBase {
   getState(el: HTMLElement): {
     label: string;
     value: string | null;
-    valueString: string[] | number | string;
+    valueString: string[] | number | string | undefined;
```

Comment:
Get rid of this

Reply:
Have a `NonNullable` equivalent but that only removes undefined, not null

Reply:
`$input.val()` should never be returning `undefined`. Have set this value below.

```ts
valueString: $input.val() as NotUndefined<ReturnType<typeof $input.val>>,
```